charts/pathfinder-rpc-server/values.yaml

```
@@ -0,0 +1,36 @@
+# LetsEncrypt certificates
+acmeEmail:                 circlesubi@gmail.com
+acmeServer:                https://acme-v02.api.letsencrypt.org/directory
+
+# Smart Contracts
+contracts:
+  hubAddress:
+
+
+
+
+# Pathfinder proxy
+pathfinderProxy:
+  port: 8675
+  upstreamServiceEndpoints: http://pathfinder2:54389
+  upstreamHealthEndpoints:  http://pathfinder2-updater:8794
+
+# Pathfinder updater
+pathfinderUpdater:
+  indexerDBConnectionString:
+  indexerWSurl:              ws://blockchain-indexer:8675
+  internalCapacityGraph:     /var/pathfinder2/data/capacity_graph.db
+  externalCapacityGraph:     /var/pathfinder2/data/capacity_graph.db
+  rpcURL:                    http://pathfinder2:54389
+
+postgresql:
+  enabled: true 
+  auth:
+    username:                 doadmin
+    database:                 index
+    existingSecret:           indexer-db-secret 
+  primary:
+    service:
+      type:                   NodePort
```

I'm not sure if NodePort is a legitimate strategy here. Do you need to expose your database to the host? As a sane default, I would suggest:

```suggestion
      type:                   ClusterIP
```

---

yes, so we could restore the database from local - unless you got another idea?

---

You can always run something like

```
kubectl -n pathfinder exec -it pod/pathfinder-postgresql-0 -- sh -c 'psql postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@localhost/${POSTGRES_DB}' < indexer.dump
```

and ingest the SQL file via `stdin`.

This frees us from the burden to open local ports on eventual cloud nodes with this Helm Chart. These values can also live as another overlay in the values of your actual deployment of the Chart.

The default values presented here would then remain as generic and agnostic as possible.

---

make sense. I will update the documentation

---

You always also have the possibility to use a `port-forward` with `kubectl`, like:

```shell
kubectl -n pathfinder port-forward svc/pathfinder-postgresql 5432:5432
```

Reference:

* [Use Port Forwarding to Access Applications in a Cluster | Kubernetes](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/)

---

actually I tried already all those three ways of loading the db but all of them threw different errors, like not letting me input the db or returning tty issues if running with `-it` . the only time it worked was with NodePort

---

Hi Jacque!
Thanks for taking this into account. From counting, I can only see two suggested alternatives to access the database via `kubectl`. Which was the third option that you were considering? I don't seem to be able to find it in the conversation above. 🙈 

The most convenient way is the `exec` path for me. Would you like to file a separate issue with a reproducer, where we can debug the observed regression, or should we do so in the Rocket Chat? I feel that specific use case may be out of scope for a more generic chart here.

We definitely don't want to assign a NodePort service when publishing the Helm chart, why I would like to see this addressed before merging.